### PR TITLE
Lassen cleanup

### PR DESCRIPTION
**Pull request by rdaly525**
Major Changes:
-adhering to Peak family_closure style
-Moved ALU to its own file
-Created a common file for constants and common types
-renamed Enums (like ALU, Mode, Cond, etc...) to have a '_t' suffix.
-removed unnecessary imports
-moved rtl_tester to its own helper file
-Added basic tests verifying magma compilation and SMT

### CODE REVIEW

**Comment by rdaly525** [comment]
Depends on https://github.com/StanfordAHA/garnet/pull/420

**Comment by rdaly525** [comment]
@rsetaluri could you take a look at the failing garnet CI? It is failing on gemstone and magma seemingly unrelated to these changes.

File: tests/_test_mapper.py [hunk 1]
```diff
@@ -233,29 +238,29 @@ def test_coreir_ops(op):
...
-        mapper.add_rr_from_description(rule)
-
-    try:
-        mapper.map_app(app)
-    except:
-        raise NotImplementedError("You probably need to regenerate rules by running the scripts/gen_rules.py script")
-
-    imap = mapper.extract_instr_map(app)
-    assert len(imap) == 1
+#@pytest.mark.parametrize("rule",Rules)
```
Comment (cdonovick) [inline tests/_test_mapper.py]: pytest.mark.skip if you want disable the test.
Remove if its not valid

Comment (rdaly525) [reply tests/_test_mapper.py]: Removed bad file and replaced it with a basic rewrite rule test for lassen

**Comment by rdaly525** [comment]
https://github.com/StanfordAHA/aha/pull/473

**Comment by rdaly525** [comment]
@cdonovick, pushed a new commit to fix with changes in peak (removing _fc methods on adt types)

**Pull request merged**